assets/src/edit-story/components/panels/textStyle/color.js

```diff
@@ -53,10 +78,62 @@ function ColorControls({ selectedElements, pushUpdate }) {
           data-testid="text.backgroundColor"
           hasGradient
           value={backgroundColor}
-          onChange={(value) => pushUpdate({ backgroundColor: value }, true)}
+          onChange={(value) =>
+            pushUpdate(
+              {
+                backgroundColor: value,
+              },
+              true
+            )
+          }
           label={__('Background color', 'web-stories')}
         />
       </Row>
+      <FillRow>
+        <FillLabel>{__('Fill', 'web-stories')}</FillLabel>
+        <ToggleButton
+          icon={<FilledIcon width={32} height={32} />} // TODO(beto): This icon should be replaced as soon as we have it on Figma
+          value={backgroundType === BACKGROUND_TEXT_MODE.NONE}
+          label={__('None', 'web-stories')}
+          onChange={(value) =>
+            pushUpdate(
+              {
+                backgroundType: value ? BACKGROUND_TEXT_MODE.NONE : '',
+              },
+              true
+            )
+          }
+        />
+        <Space flex="0 0 15px" />
+        <ToggleButton
+          icon={<FilledIcon width={32} height={32} />}
+          value={backgroundType === BACKGROUND_TEXT_MODE.FILL}
+          label={__('Fill', 'web-stories')}
+          onChange={(value) =>
+            pushUpdate(
+              {
+                backgroundType: value ? BACKGROUND_TEXT_MODE.FILL : '',
+              },
+              true
+            )
+          }
+        />
+        <Space flex="0 0 15px" />
+        <ToggleButton
+          icon={<HighlightedIcon width={32} height={32} />}
+          label={__('Highlight', 'web-stories')}
+          value={backgroundType === BACKGROUND_TEXT_MODE.HIGHLIGHT}
+          onChange={(value) =>
+            pushUpdate(
+              {
+                backgroundType: value ? BACKGROUND_TEXT_MODE.HIGHLIGHT : '',
```

Comment:
Does the empty string actually have meaning for us? Or is it always `BACKGROUND_TEXT_MODE.NONE`? If so, let's use the `BACKGROUND_TEXT_MODE.NONE` value? Alternatively it might be cleaner to do update on the `true` value only. E.g.:

```
onChange({value} => value && pushUpdate({backgroundMediaMode: BACKGROUND_TEXT_MODE.HIGHLIGHT}, true))
```